### ISSUE TITLE
[functional testing] Remove time.sleeps where possible

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#3486]
In the existing Tor Browser functional testing branch, `time.sleep`s are introduced to wait for elements to appear, etc. We should use `FunctionalTest.wait_for` where needed (and increase the timeout as needed since we're testing over Tor). Otherwise, the overall test time may be unnecessarily lengthened.  

Epic: #3659

**Comment by kushaldas** [comment]
The reason those `time.sleep`s got introduced was increased time for page loads. #3484 is also one thing that vanishes if I have a large `time.sleep` value (10 seconds).

**Comment by kushaldas** [comment]
If we start adding assertion based `wait_for` calls for each of the mouse clicks (along with a long time out, say 30 seconds), it seems things are passing. But, this will increase the lines of code in the functional tests side a lot (that should be okay). 

On the other hand, I still see a few failures while connecting to the Tor network, that is just required while testing against external `.onion` addresses.

**Comment by kushaldas** [comment]
Pushing `WIP` code into my https://github.com/kushaldas/securedrop/tree/add_all_asserts_of_the_world branch. Will keep updating that.

**Comment by kushaldas** [comment]
```
./bin/dev-shell ./bin/run-test --capture=no -v tests/functional/test_journalist.py::TestJournalist::test_journalist_interface_ui_with_modal
```

This test is failing in my branch at the last function call `_journalist_uses_js_buttons_to_download_unread`.

**Comment by kushaldas** [comment]
I have updated my branch. I can see some errors (randomly sometimes) in https://github.com/kushaldas/securedrop/blob/add_all_asserts_of_the_world/securedrop/tests/functional/journalist_navigation_steps.py#L272 while running `test_admin_interface.py`.

@redshiftzero please have a look if you find some time.

**Comment by redshiftzero** [comment]
I tried this branch today against a staging server (just `./bin/dev-shell ./bin/run-test tests/functional/test_admin_interface.py` since you reported issues) and I am also seeing errors, but I ran into slightly different problems: for me this button was not reliably clicked:   https://github.com/kushaldas/securedrop/blob/add_all_asserts_of_the_world/securedrop/tests/functional/journalist_navigation_steps.py#L314 - have you seen this before? 

I do recall in the past we've had issues with Selenium reliably clicking (e.g. sometimes we've used the enter key instead of clicking for this reason), I suspect that might also be causing the issue you ran into (I'm assuming there is a timeout on the assertion [here](https://github.com/kushaldas/securedrop/blob/add_all_asserts_of_the_world/securedrop/tests/functional/journalist_navigation_steps.py#L269)?).

**Comment by redshiftzero** [comment]
Closed by #3870
